recipes/cppunit/all/conanfile.py

```diff
@@ -72,10 +67,12 @@ def generate(self):
             tc.extra_defines.append("CPPUNIT_BUILD_DLL")
         if is_msvc(self):
             tc.extra_cxxflags.append("-EHsc")
-            if (self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) >= "12") or \
-               (self.settings.compiler == "msvc" and Version(self.settings.compiler.version) >= "180"):
+            if (str(self.settings.compiler) == "Visual Studio" and Version(self.settings.compiler.version) >= "12") or \
+               (str(self.settings.compiler) == "msvc" and Version(self.settings.compiler.version) >= "180"):
                 tc.extra_cflags.append("-FS")
                 tc.extra_cxxflags.append("-FS")
+        if is_apple_os(self):
+            tc.extra_ldflags.append("-headerpad_max_install_names")
```

Comment:
https://github.com/conan-io/conan-center-index/pull/13892#issuecomment-1302377019 @jcar87 I am not the expert but I know this is sus, maybe take a peek here?

Reply:
See https://github.com/conan-io/conan-center-index/pull/15759#issuecomment-1419046535

Reply:
If the particular library needs it for a particular reason - please add a comment in the recipe justifying the need for this flag, and add the flag accordingly. We have tested this locally and cannot reproduce this on newer versions of Xcode - probably something to do with Xcode 11.0.


In an ideal world, libtool would support relative rpaths on macos, and we would have a better way of handling this rather than patching things post-install like we do with `fix_apple_shared_install_name`. My understanding is that so far libtool has not been updated for this: https://lists.gnu.org/archive/html/libtool-patches/2010-09/msg00308.html 

That said, if there were widespread issues with this, we would consider a different approach - but in the 63 files I see that make a call to `fix_apple_shared_install_name`, this is the first instance where we have seen it not working.

Reply:
```suggestion
        if is_apple_os(self):
            # https://github.com/conan-io/conan-center-index/pull/15759#issuecomment-1419046535
            tc.extra_ldflags.append("-headerpad_max_install_names")
```